File: Serene/Serene.Script/Imports/FormContexts/Northwind.ShipperForm.cs

namespace Serene.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Runtime.CompilerServices;

    public partial class ShipperForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Northwind.Shipper";
    
        public ShipperForm(string idPrefix) : base(idPrefix) {}
    
        public StringEditor CompanyName { [InlineCode("{this}.w('CompanyName', Serenity.StringEditor)")] get; private set; }
        public PhoneEditor Phone { [InlineCode("{this}.w('Phone', Serene.Northwind.PhoneEditor)")] get; private set; }
    }
}

File: Serene/Serene.Web/Modules/Common/Imports/ServerTypings/Northwind.ShipperForm.ts
namespace Serene.Northwind {
    export class ShipperForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.Shipper';

    }

    export interface ShipperForm extends Serenity.PrefixedContext {
        CompanyName(): Serenity.StringEditor;
        Phone(): PhoneEditor;
    }

    [['CompanyName', Serenity.StringEditor], ['Phone', PhoneEditor]].forEach(x => ShipperForm.prototype[<string>x[0]] = function() { return this.w(x[0], x[1]); });
}

